src/helpers/clipboard/clipboard_figure_state.ts

```diff
@@ -84,14 +84,19 @@ export class ClipboardFigureState implements ClipboardState {
    */
   paste(target: Zone[]) {
     const sheetId = this.getters.getActiveSheetId();
+    const { width, height } = this.copiedFigure;
+    const numCols = this.getters.getNumberCols(sheetId);
+    const numRows = this.getters.getNumberRows(sheetId);
+    const { start: colStart } = this.getters.getColDimensions(sheetId, target[0].left);
+    const { start: rowStart } = this.getters.getRowDimensions(sheetId, target[0].top);
+    const { end: lastColEnd } = this.getters.getColDimensions(sheetId, numCols - 1);
+    const { end: lastRowEnd } = this.getters.getRowDimensions(sheetId, numRows - 1);
     const position = {
-      x: this.getters.getColDimensions(sheetId, target[0].left).start,
-      y: this.getters.getRowDimensions(sheetId, target[0].top).start,
+      x: lastColEnd - (colStart + width) < 0 ? lastColEnd - width : colStart,
```

Comment:
What is the expected behavior when width is bigger than lasColEnd ?

You can imagine the following scenario:;
I have 2 sheets
in the second sheet, I delete all but 1 column (only have col A)
in the first sheet I create a chart, and make it large enough
copy the chart
got to sheet2, paste it

the current code will dispatch negative positions for x (you can do the same reasoning with the rows)

I think we want x to be non-negative. Could you check this with our PO ?